storage/pruning/pruningStorer.go

```diff
@@ -523,10 +523,10 @@ func (ps *PruningStorer) GetBulkFromEpoch(keys [][]byte, epoch uint32) (map[stri
 			continue
 		}
 
-		returnMap[string(key)] = res
+		results = append(results, storage.KeyValuePair{Key: key, Value: res})
```

Comment:
regarding the duplications from the PR description, we cannot have any duplicates this way because the storer will contain unique keys. Therefore, I don't see any case where we can have duplicates

Reply:
Previously, slightly-corrupted input (duplicated requested keys) resulted into non-corrupted output (duplicated requested keys removed). Now, duplicated requested keys would result into duplicated response items (even though the `storer` cannot have duplicate keys).

Reply:
I see. I think that the current code is ok. Our code that requests txs will not call for duplicated, so we're fine. No need to further filter the transactions